include/dlaf/matrix/matrix.h

```diff
@@ -297,23 +306,36 @@ class Matrix<const T, D> : public internal::MatrixBase {
     done(distribution().local_tile_index(index));
   }
 
-protected:
   Matrix(Distribution distribution) : internal::MatrixBase{std::move(distribution)} {
     DLAF_ASSERT((distribution.offset() == GlobalElementIndex{0, 0}), "not supported",
                 distribution.offset());
   }
+
+  template <class MatrixLike>
+  Matrix(MatrixLike& mat, const LocalTileSize& tiles_per_block) noexcept
+      : MatrixBase(mat.distribution(), tiles_per_block) {
+    setUpRetiledSubPipelines(mat, tiles_per_block);
+  }
```

Comment:
I'm going to resolve this conversation because it appears that the comment time is reported as "last week" and the code is also marked as "outdated". I already made this constructor protected again, as part of the big cleanup last week (making it public was for initial testing purposes).